specs/templates/table-column-hld.md

```diff
@@ -0,0 +1,134 @@
+# Table Column [Template]
+
+## Overview
+
+*A high-level description of the column.*
+
+### Background
+
+- *Relevant historical or background information*
+- *Link to relevant work items, related existing issues, etc.*
+
+### Features
+
+*A list of the key features unique to this component.*
+
+### Non-goals
+
+*A list of use cases, features, or functionality which are **not** goals for the component.*
```

Comment:
Here are a couple ideas for additional content/instructions to incorporate. I think they're both already covered at a high level by sections in the template but it might be useful to augment some of the instructions as a reminder of the level of detail we're looking for. Just a suggestion though; your call if you can fit them in succinctly.

1. A common reason to create a new column would be to expose capability of an existing nimble component within the table. When we do that, we should evaluate each of the existing component's APIs and decide if they should be configurable at the row level, column level, or not at all. Could maybe request that those APIs be listed explicitly in "Non-goals" and "Column Element"? 
2. Every column should have 1 or more `xxx-field-name` attributes that populate per-row state from the data. Should we call that out specifically in "Column Element" or will it fall out naturally?

Reply:
> Every column should have 1 or more xxx-field-name attributes that populate per-row state from the data. Should we call that out specifically in "Column Element" or will it fall out naturally?

I think that will fall out naturally. And in some rare cases, a column might not have such an attribute (like the button column I was proposing).